### DISCUSSION TITLE
Custom form in Buttondown subscribe page? (instead of the iframe form)

### DISCUSSION THREAD

**Discussion by christophberger** [buttondown/discussions#15] [Ideas]
I created a custom HTML form with a checkbox and a privacy policy link, for GDPR reasons.
To add the checkbox, I created a custom field in the Buttondown settings, and in the generated HTML, I changed the type to checkbox. 

This works well in the HTML form. The checkbox status gets saved to the subscriber's metadata.

 
![Screenshot 2024-02-01 at 18 27 18](https://github.com/buttondown-email/discussions/assets/3835890/f2a454bf-f2af-4705-89a1-2b2c51e80ed6)

However, my Buttondown landing page (buttondown.email/christophberger) uses the iframe version of the subscribe form. The iframe version is not configurable, and the checkbox still shows as a plain text field. 

![Screenshot 2024-02-01 at 18 27 02](https://github.com/buttondown-email/discussions/assets/3835890/48872374-c09e-4541-b7f2-0d159781371e)

Is there a way of
- either replacing the iframe form by a custom HTML form on that page,
- or changing the type of the custom input field to checkbox for the iframe version, and adding extra text?

**Comment by jmduke** [comment]
This will be solved by https://github.com/buttondown-email/roadmap/issues/772 (which is poorly named but does exactly what you want; lets those fields get manifested as checkboxes.) I'll have it shipped by this time next week!

**Reply by christophberger** [reply]
Awesome! For now, I think I can keep people away from the signup page. (I launched my custom HTML forms on my site just a few seconds ago.)
